### PR TITLE
Add implementation of SVDF kernel.

### PR DESCRIPTION
**Pull request by jenselofsson**
Optimized version of the SVDF-kernel compatible with TensorFlow Lite Micro.

### CODE REVIEW

**Comment by jenselofsson** [comment]
Fixed in https://github.com/tensorflow/tensorflow/pull/42651

File: CMSIS/NN/Source/SVDFunctions/arm_svdf_s8.c [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+      // Process the remaining data
+      for (; j < time_batches; j++)
+      {
+        sum += *v1 * *v2;
+        v1++;
+        v2++;
+      }
+#else
+      int32_t sum = 0;
```
Comment (felix-johnny) [inline CMSIS/NN/Source/SVDFunctions/arm_svdf_s8.c:185]: Please fix the redefinition of sum.

File: CMSIS/NN/Source/SVDFunctions/arm_svdf_s8.c [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  const int16_t rank = svdf_params->rank;
+
+  int32_t zp_32 = (-zp_in & 0xffff) |
+                 ((-zp_in & 0xffff) << 16);
+
+  const int32_t input_batches = input_dims->n;
+  const int32_t input_height = input_dims->h;
+  const int32_t feature_batches = weights_feature_dims->n;
+  const int32_t time_batches = weights_time_dims->h;
+  const int32_t unit_cnt = feature_batches / rank;
```
Comment (felix-johnny) [inline CMSIS/NN/Source/SVDFunctions/arm_svdf_s8.c:93]: please use count here instead of cnt

**Review by felix-johnny — CHANGES_REQUESTED** [review]

File: CMSIS/NN/Source/SVDFunctions/arm_svdf_s8.c (hunk 1 above)
Comment (jenselofsson) [reply CMSIS/NN/Source/SVDFunctions/arm_svdf_s8.c:185]: Done.

File: CMSIS/NN/Source/SVDFunctions/arm_svdf_s8.c (hunk 2 above)
Comment (jenselofsson) [reply CMSIS/NN/Source/SVDFunctions/arm_svdf_s8.c:93]: Done.

File: CMSIS/NN/Include/arm_nnfunctions.h [hunk 3]
```diff
@@ -2061,6 +2062,61 @@ void arm_softmax_u8(const uint8_t *input,
...
+ *
+ */
+
+    /**
+     * @brief s8 SVDF function
+     *
+     * @param[in]   input_ctx Temporary scratch buffer
+     * @param[in]   output_ctx Temporary output scratch buffer
+     * @param[in]   svdf_params SVDF Parameters
+     *                          Range of svdf_params->input_offset  : [-127, 128]
```
[the remark this answers is not in the document]
Comment (jenselofsson) [reply CMSIS/NN/Include/arm_nnfunctions.h:2076]: Correct, when I double check it I realize I've simply mistyped it.

File: CMSIS/NN/Include/arm_nnfunctions.h [hunk 4]
```diff
@@ -2061,6 +2062,61 @@ void arm_softmax_u8(const uint8_t *input,
...
+     *                          Range of svdf_params->output_offset  : [-127, 128]
+     * @param[in]   input_quant_params Input quantization parameters
```
[the remark this answers is not in the document]
Comment (jenselofsson) [reply CMSIS/NN/Include/arm_nnfunctions.h:2078]: The "Range of svdf_params..." describes the svdf_params argument. Should I the two lines above line 2075 instead?

**Review by felix-johnny — APPROVED** [review]

**Pull request merged**